packages/flutter/test/widgets/dismissible_test.dart

```diff
@@ -143,6 +147,58 @@ Future<Null> dismissItem(WidgetTester tester, int item, {
   await tester.pump(); // rebuild after the callback removes the entry
 }
 
+Future<Null> rollbackElement(WidgetTester tester, Finder finder, { @required AxisDirection gestureDirection }) async {
+  Offset downLocation;
+  Offset upLocation;
+  switch (gestureDirection) {
+    case AxisDirection.left:
+      // getTopRight() returns a point that's just beyond itemWidget's right
+      // edge and outside the Dismissible event listener's bounds. Just nudge
+      // the itemWidget before it's threshold.
+      downLocation = tester.getTopRight(finder) + const Offset(-0.1, 0.0);
+      upLocation = tester.getTopRight(finder) + const Offset(-0.2, 0.0);
+      break;
+    case AxisDirection.right:
+      // we do the same thing here to keep the test symmetric
+      downLocation = tester.getTopLeft(finder) + const Offset(0.1, 0.0);
+      upLocation = tester.getTopLeft(finder) + const Offset(0.2, 0.0);
+      break;
+    case AxisDirection.up:
+      // getBottomLeft() returns a point that's just below itemWidget's bottom
+      // edge and outside the Dismissible event listener's bounds.
+      downLocation = tester.getBottomLeft(finder) + const Offset(0.0, -0.1);
+      upLocation = tester.getBottomLeft(finder) + const Offset(0.0, -0.2);
+      break;
+    case AxisDirection.down:
+      // again with doing the same here for symmetry
+      downLocation = tester.getTopLeft(finder) + const Offset(0.0, 0.1);
+      upLocation = tester.getTopLeft(finder) + const Offset(0.0, 0.2);
+      break;
+    default:
```

Comment:
you can remove the default; the analyzer will then warn us at analyze time if there's an unsupported gesture direction